operator/scheduler/utils.go

```diff
@@ -35,34 +35,42 @@ func handleLoadedExperiments(
 		client.InNamespace(namespace),
 	)
 	if err != nil {
-		return
+		return err
 	}
 
 	for _, experiment := range experimentList.Items {
 		// experiments that are not in the process of being deleted has DeletionTimestamp as zero
 		if experiment.ObjectMeta.DeletionTimestamp.IsZero() {
 			s.logger.V(1).Info("Calling start experiment (on reconnect)", "experiment", experiment.Name)
-			if _, err := s.StartExperiment(ctx, &experiment, grpcClient); err != nil {
-				// if this is a retryable error, we will retry on the next connection reconnect
+			if retryable, err := s.StartExperiment(ctx, &experiment, grpcClient); err != nil {
 				s.logger.Error(err, "Failed to call start experiment", "experiment", experiment.Name)
+				if retryable {
+					// if this is a retryable error, we will retry on the next connection reconnect
+					return err
```

Comment:
If I understand things correctly, perhaps the comment here is slightly misleading: this will not only retry on the next connection reconnect, but will actually stop all the other object states from being sync-ed to the scheduler and trigger the controller to resubscribe to the scheduler.

Reply:
yes will reflect.